EIPS/eip-1052.md

```diff
@@ -18,16 +18,47 @@ Many contracts need to perform checks on a contract's bytecode, but do not neces
 Contracts can presently do this using the `EXTCODECOPY` opcode, but this is expensive, especially for large contracts, in cases where only the hash is required. As a result, we propose a new opcode, `EXTCODEHASH`, which returns the keccak256 hash of a contract's bytecode.
 
 ## Specification
-A new opcode, `EXTCODEHASH`, is introduced, with number 0x3D. `EXTCODEHASH` takes one argument from the stack, and pushes the keccak256 hash of the code at that address to the stack.
+
+A new opcode, `EXTCODEHASH`, is introduced, with number 0x3D. The `EXTCODEHASH` 
+takes one argument from the stack, zeros first 12 bytes of the taken value by
+performing modulo operation by 2<sup>160</sup> to obtain an account address.
+Then, it pushes the keccak256 hash of the code of the account and the address 
+to the stack or `0` in case the account does not exist.
```

Comment:
What does it push if the account exists but has no code?

Reply:
The keccak256 hash of empty data, i.e. `c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470`. Should that be explained?

Reply:
I think it'd be a good idea, yes.

Reply:
I added additional sentence about this. Also added a Rationale entry. But I'm not 100% confident this is good design.